progetto_esa.tex

```diff
@@ -61,10 +60,32 @@ \section{Basic Configuration}
 The command to start the configuration process is: 
 make menuconfig.
 This tools is based in keyconfig, that has the task to reads all the features available to build our OS.
-The keyconfig reads all the config.in files contained in each package available.
-The folders of the pcackages contain the config.in file and the package.mk, the .mk file contains the instruction to compile the package for buildroot.
+The keyconfig reads all the Config.in files contained in each package available.
+The folders of the pcackages contain the Config.in file and the package.mk, the .mk file contains the instruction to compile the package for buildroot.
 Keyconfig builds the tree of the configuration menu (menuconfig) so that we are able to select all the features that we would include.
-It can also allow to change things like hostname, init system, root password, in System configuration or select a specific kernel in Kernel section.
+In most of the cases that we would like to configure and compile build root, we have defconfig files that make life easier.
+Those files exist for real world and virtual devices and have the basic information already defined to make the configuration process faster.
+Def\_config files are invoked before that we are going to start the menuconfig tool, so that the configuration is almost ready for our specific real or virtual platform.
+To invoke a Def\_config file we simply run the command
+make Def\_configfile
+and then, we are ready to start the menu\_config to customize our distribution parameters.
+
+We are now ready to give a brief description of the main section that we have in this configuration tool.
+\begin{itemize}
+\item The Build Options submenu selects options that Buildroot will use to download and build the packages and files needed by the host and the target.
+\item The System configuration submenu selects options for the system such as which INIT to run, root login, root password, host name and login prompt. It also supports scripts that may be run during the build.
+\item The Kernel submenu selects the Linux kernel, the location to download the kernel source and options in the build process.
+\item The Target Packages submenu selects from over 2000+ software packages. The submenu lists
+them by category. For example, Audio and Video, Games, Networking Applications, etc.
+\item The Filesystem Images submenu selects the filesystem type and options.
+\item The Legacy config options submenu allows the developer to select software that has been removed in the present version of Buildroot. This submenu makes it convenient for the developer to maintain backward compatibility.
+\end{itemize}
+At this point, if you decide to save your configuration to invoke it for the future use, you just need to issue the command:
+make savedefconfig
```

Comment:
make savedefconfig will override the defconfig you started from: if you want to have a new defconfig with your own options enabled, you need to issue "make savedefconfig BR2_DEFCONFIG=<path-to-defconfig>"